corehq/ex-submodules/casexml/apps/case/cleanup.py

```diff
@@ -136,14 +137,24 @@ def claim_case(domain, restore_user, host_id, host_type=None, host_name=None, de
     return claim_id
 
 
-def get_first_claim(domain, user_id, case_id):
+def get_first_claims(domain, user_id, case_ids):
     """
-    Returns the first claim by user_id of case_id, or None
+    Returns the first claim by user_id of case_ids, or None
     """
-    case = CommCareCase.objects.get_case(case_id, domain)
+    cases_found = CommCareCase.objects.get_cases(case_ids, domain)
+    case_ids_found = [case.get_id for case in cases_found]
+    cases_not_found = [case for case in case_ids if case not in case_ids_found]
+
+    if len(cases_not_found) != 0:
+        raise CaseNotFound(", ".join(cases_not_found))
+
     identifier = DEFAULT_CASE_INDEX_IDENTIFIERS[CASE_INDEX_EXTENSION]
-    try:
-        return next((c for c in case.get_subcases(identifier)
-                     if c.type == CLAIM_CASE_TYPE and c.owner_id == user_id and c.closed is False))
-    except StopIteration:
-        return None
+    previously_claimed_ids = list()
+    for case in cases_found:
+        try:
+            if next((c for c in case.get_subcases(identifier)
+                    if c.type == CLAIM_CASE_TYPE and c.owner_id == user_id and c.closed is False)):
+                previously_claimed_ids.append(case.get_id)
```

Comment:
I read this a few times thinking it was an indentation error. Could you pull the `c for c...` expression into a local variable before the loop?